### PR TITLE
Disable the vs30_tolerance check on-demand

### PR DESCRIPTION
**Pull request by ptormene**
Fixes https://github.com/gem/oq-engine/issues/9377

### CODE REVIEW

File: openquake/hazardlib/site_amplification.py [hunk 1]
```diff
@@ -261,6 +261,13 @@ def check(self, vs30, vs30_tolerance, gsims_by_trt):
         """
         for gsims in gsims_by_trt.values():
             for gsim in gsims:
+                if vs30_tolerance < 0:
+                    continue
+                elif not hasattr(gsim, 'DEFINED_FOR_REFERENCE_VELOCITY'):
+                    raise AttributeError(
+                        f'The parameter "vs30_tolerance" is >= 0, but'
+                        f' the attribute DEFINED_FOR_REFERENCE_VELOCITY is'
```
Comment (micheles) [inline openquake/hazardlib/site_amplification.py:269]: I would write
```
The attribute DEFINED_FOR_REFERENCE_VELOCITY is missing in the gsim {gsim}.
However, at your peril, you can disable the vs30 consistency check by setting
vs30_tolerance = -1
```

File: openquake/hazardlib/tests/site_amplification_test.py [hunk 2]
```diff
@@ -102,6 +102,20 @@ class AmplifierTestCase(unittest.TestCase):
...
+                      index='ampcode')
+        a = Amplifier(self.imtls, df, self.soil_levels)
+        gmm = valid.gsim('CanadaSHM6_ActiveCrust_BooreEtAl2014')
+        with self.assertRaises(AttributeError) as ctx:
+            a.check(self.vs30, 0, {TRT.ACTIVE_SHALLOW_CRUST: [gmm]})
+        self.assertIn('The parameter "vs30_tolerance" is >= 0, but'
+                      ' the attribute DEFINED_FOR_REFERENCE_VELOCITY'
+                      ' is missing in the gsim'
+                      ' [CanadaSHM6_ActiveCrust_BooreEtAl2014]',
+                      str(ctx.exception))
```
Comment (micheles) [inline openquake/hazardlib/tests/site_amplification_test.py:117]: You also need a test with CanadaSHM6_ActiveCrust_BooreEtAl2014 and vs30_tolerance=-1.

File: openquake/hazardlib/site_amplification.py (hunk 1 above)
Comment (ptormene) [reply openquake/hazardlib/site_amplification.py]: Done

File: openquake/hazardlib/tests/site_amplification_test.py [hunk 3]
```diff
@@ -102,6 +102,29 @@ class AmplifierTestCase(unittest.TestCase):
...
+            a.check(self.vs30, 0, {TRT.ACTIVE_SHALLOW_CRUST: [gmm]})
+        self.assertIn(
+            'The attribute DEFINED_FOR_REFERENCE_VELOCITY is missing in the'
+            ' gsim [CanadaSHM6_ActiveCrust_BooreEtAl2014]. However, at your'
+            ' peril, you can disable the vs30 consistency check by setting'
+            ' vs30_tolerance = -1',
+            str(ctx.exception))
+
+    def test_missing_attribute_but_negative_vs30_tolerance(self):
+        fname = gettemp(trivial_ampl_func)
```
Comment (micheles) [inline openquake/hazardlib/tests/site_amplification_test.py:121]: You can put this test inside the one above to avoid reading twice the same file.

Comment (ptormene) [reply openquake/hazardlib/tests/site_amplification_test.py]: Done

**Review by micheles — APPROVED** [review]

**Pull request merged**